telemetry/definitions/commonDefinitions.json

```diff
@@ -1067,6 +1067,50 @@
                 { "type": "ec2ConnectionType" }
             ]
         },
+        {
+            "name": "ec2_createAmi",
+            "description": "Create an image from an EC2 Instance",
+            "metadata": [
+                { "type": "result" }
+            ]
+        },
+        {
+            "name": "ec2_editInstanceElasticIp",
+            "description": "Associate or disassociate an Elastic IP with an EC2 Instance",
+            "metadata": [
+                { "type": "result" }, 
+                { "type": "enabled", "required": false}
```

Comment:
So was this `enabled` field one of the reasons why we didn't want generic definitions? I could see how this might clash with something generic.

Reply:
Later on we could realize that we want additional properties on specific events. By not going the generic metric approach, we avoid having a metric containing a higher list of (optional) metadata fields.

Reply:
> Later on we could realize that we want additional properties on specific events

This can still be done with generic events due to how we define metadata but it's not exactly intuitive for developers. It doesn't require doing anything special, it's just a different way of thinking about things.